pkg/driver/controller_server.go

```diff
@@ -864,6 +865,20 @@ func (driver *Driver) controllerPublishVolume(
 	publishContext[targetNamesKey] = strings.Join(publishInfo.AccessInfo.BlockDeviceAccessInfo.TargetNames, ",")
 	publishContext[targetScopeKey] = requestedTargetScope
 	publishContext[lunIDKey] = strconv.Itoa(int(publishInfo.AccessInfo.BlockDeviceAccessInfo.LunID))
+
+	// Start of population of target array details
+	log.Tracef("\n PUBLISH INFO ::::: %v", publishInfo)
+
+	if publishInfo.AccessInfo.BlockDeviceAccessInfo.SecondaryBackendDetails.PeerArrayDetails != nil {
+		secondaryArrayMarshalledStr, err := json.Marshal(&publishInfo.AccessInfo.BlockDeviceAccessInfo.SecondaryBackendDetails)
+		if err != nil {
+			log.Errorf("Error in marshalling secondary details %s", err.Error())
```

Comment:
i think we should fail publish request, if there is an error here as we cannot connect to secondary array without these in this case.

Reply:
I added code to return nil, error response.